highlight/highlight
# feat: Support simple password authentication for the self-hosted deploy

## Summary

Implemented a new Authentication mode for backend/frontend to allow anyone deploying the application to 
set a password in their environment that may be shared with "authorized" users.

## IMPLEMENTATION Details

## Frontend
- Support Password auth mode
- implement Password Class in frontend/src/util/auth.tsx end ensure it is used when REACT_APP_AUTH_MODE=password
  - This is similar to simple auth except that it makes a request to our public endpoint to "authenticate the" credentials given.
- On success an authorization token is set and UI works as expected 
- Add auth token to request header as is the case with firebase auth

## Backend
- implement an endpoint at /private/login to "authenticate" user using our set ADMIN_PASSWORD
- On success this endpoint returns a token and user data
- this middleware just ensures that the app may be authorized to use "protected" "endpoints" if it has a valid
   authorization token issued by us.

## Loom Demo
https://www.loom.com/share/4b5485fdfcd14805a9659c2eabe4f94e?sid=2bd72638-3584-462a-b79f-4f92861c9768


Fixes #6483
/claim #6483



## How did you test this change?
- SET REACT_APP_AUTH_MODE=password in ".env"
- SET ADMIN_PASSWORD in docker/.env
- RUN the application with docker using the ./run.sh script in the docker folder
- Visit https://localhost:3000/sign_in
- Try logging in with an incorrect(not equal to ADMIN_PASSWORD) password and observe that it fails
- Try logging in with admin password and observe a successful login with any email but using the ADMIN_PASSWORD set in the previous steps succeeds

## FURTHER TESTING (TO ENSURE SimpleAuth is not broken)
- Reset REACT_APP_AUTH_MODE=simple (NO AUTH)
- Observe that you are now able to login without credentials as was the case for simple auth



## Are there any deployment considerations?
A new env variable called ADMIN_PASSWORD has been introduced. This is not relevant if you do not intend to use
password auth mode. 

A new option ('password') for the existing REACT_APP_AUTH_MODE environment variable has been added.
So you can now set REACT_APP_AUTH_MODE=password along with a relevant value for ADMIN_PASSWORD
to support this feature


There are no migrations

## Does this work require review from our design team?
No. There are no design changes
